tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -3227,7 +3227,8 @@ LaunchDimensions IrEmitterUnnested::EmitKernel(
           });
 
       // Wait for all threads to reach this point using `__syncthreads` in CUDA.
-      EmitCallToTargetIntrinsic(TargetIntrinsicID::kBarrierId, {}, {}, &b_);
+      EmitCallToTargetFunction(TargetFunctionID::kBarrierId, {}, {},
+                               PRIMITIVE_TYPE_INVALID, {}, {}, &b_);
```

Comment:
Using PRIMITIVE_TYPE_INVALID to represent "void" seems wrong to me.  I think there should be a separate overload for void functions.